packages/core/src/testE2E/amazonq/framework/messenger.ts

```diff
@@ -73,29 +74,48 @@ export class Messenger {
         return [...(this.getStore().chatItems ?? [])]
     }
 
+    getPlaceholder() {
+        return this.getStore().promptInputPlaceholder
+    }
+
+    hasButton(type: FollowUpTypes) {
+        return this.getChatItems()
+            .pop()
+            ?.followUp?.options?.map(opt => opt.type)
+            .includes(type)
```

Comment:
Am I missing something? This code either just throws TypeError on `undefined` or returns `undefined`, probably the former, because the `.includes` isn't chained

Can we make this method always return boolean? I don't really see the point of using this method as below with `this.hasButton(FollowUpTypes.Retry) ?? false`